src/context.rs

```diff
@@ -120,3 +134,74 @@ mod std_only {
     }
 
 }
+
+impl<'buf> Signing for SignOnlyPreallocated<'buf> {}
+impl<'buf> Signing for AllPreallocated<'buf> {}
+
+impl<'buf> Verification for VerifyOnlyPreallocated<'buf> {}
+impl<'buf> Verification for AllPreallocated<'buf> {}
+
+unsafe impl<'buf> Context for SignOnlyPreallocated<'buf> {
+    const FLAGS: c_uint = ffi::SECP256K1_START_SIGN;
+    const DESCRIPTION: &'static str = "signing only";
+
+    fn deallocate(ptr: *mut [u8]) {
+        let _ = ptr;
```

Comment:
I think it'd be clearer to write
```
fn deallocate(_: *mut [u8]) {
    // do nothing
}
```
I read `let _ = ptr` as "drop `ptr`" which is technically what's happening but it's a bit weird because dropping a pointer is a noop.